runtime/callback.c

```diff
@@ -27,6 +27,25 @@
 
 static __thread int callback_depth = 0;
 
+/*
+ * These macros are for ensuring effects are handled correctly
+ * inside callbacks. There are two aspects:
+ *  - we clear the stack parent for a callback to force an Unhandled
+ *  exception rather than effects being passed over the callback
+ *  - we register the stack parent as a local root while the callback
+ * is executing to ensure that the garbage collector follows the
+ * stack parent
+ */
+#define SAVE_AND_CLEAR_STACK_PARENT(cont, domain_state) \
```

Comment:
Could we not turn this into a static inline function that returns `cont`? We would use this as:

```
cont = save_and_clear_stack_parent(domain_state);
```

Reply:
that's much better. did that.